operations/scheduler.go

```diff
@@ -0,0 +1,94 @@
+package operations
+
+import (
+	"context"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/storage"
+)
+
+const scheduleMsg = "Scheduling %s job for operation with id (%s)"
+
+// DefaultScheduler implements JobScheduler interface. It's responsible for
+// storing C/U/D jobs so that a worker pool can eventually start consuming these jobs
+type DefaultScheduler struct {
+	repository storage.Repository
+	workerPool *WorkerPool
+	jobs       chan ExecutableJob
+}
+
+// NewScheduler constructs a DefaultScheduler
+func NewScheduler(smCtx context.Context, repository storage.Repository, poolSize int) *DefaultScheduler {
+	workerPool := NewWorkerPool(smCtx, repository, poolSize)
+
+	return &DefaultScheduler{
+		repository: repository,
+		workerPool: workerPool,
+		jobs:       workerPool.jobs,
+	}
+}
+
+// Run starts the DefaultScheduler's worker pool enabling him to start polling
+// for scheduled jobs
+func (ds *DefaultScheduler) Run() {
+	ds.workerPool.Run()
+}
+
+// SchedulerCreate schedules a Create job in the worker pool
+func (ds *DefaultScheduler) ScheduleCreate(ctx context.Context, object types.Object, operationID string) {
+	reqCtxClone := util.RequestContextCopy{Context: ctx}
```

Comment:
Context still gets canceled as soon as the original request context gets canceled..